ctk/common/src/main/kotlin/org/codice/compliance/verification/core/internal/StatementVerifier.kt

```diff
@@ -49,141 +44,111 @@ internal class StatementVerifier(val node: Node) {
-                        parent = AUTHN_STATEMENT,
-                        node = node)
-        }
     }
 
-    /** 2.7.3.2 Element <EncryptedAttribute> **/
-    private fun verifyEncryptedAttribute() {
-        node.allChildren(ENCRYPTED_ATTRIBUTE).forEach {
-            val encryptedData = it.children("EncryptedData")
-            if (encryptedData.isEmpty())
-                throw SAMLComplianceException.createWithXmlPropertyReqMessage("SAMLCore.2.7.3.2",
-                        property = "EncryptedData",
-                        parent = ENCRYPTED_ATTRIBUTE,
-                        node = node)
-
-            if (encryptedData
-                            .filter { it.attributes.getNamedItem(TYPE) != null }
-                            .any { it.attributes.getNamedItem(TYPE).textContent !=
-                                    TestCommon.ELEMENT })
-                throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_2_7_3_2_a,
-                        property = TYPE,
-                        actual = it.attributes.getNamedItem(TYPE).textContent,
-                        expected = TestCommon.ELEMENT,
-                        node = node)
-        }
-    }
-
-    /** 2.7.3.1 Element <Attribute> **/
-    @Suppress("ComplexCondition")
-    private fun verifyAttribute() {
-        node.allChildren(ATTRIBUTE).forEach {
-            if (it.attributes.getNamedItem("Name") == null)
-                throw SAMLComplianceException.createWithXmlPropertyReqMessage("SAMLCore.2.7.3.1",
-                        property = "Name",
-                        parent = ATTRIBUTE,
-                        node = node)
-
-            val nameAttribute = it.attributes.getNamedItem("Name")
-            val nameFormatAttribute = it.attributes.getNamedItem("NameFormat")
-            val friendlyNameAttr = it.attributes.getNamedItem("FriendlyName")
+    /** 2.7.2.1 Element <SubjectLocality> **/
+    private fun verifySubjectLocality() {
+        node.allChildren("SubjectLocality").forEach {
+            val address = it.attributes.getNamedItem("Address")
+            val dNSName = it.attributes.getNamedItem("DNSName")
 
-            if ((nameAttribute != null && nameAttribute.textContent == null)
-                    || (nameFormatAttribute != null && nameFormatAttribute.textContent == null)
-                    || (friendlyNameAttr != null && friendlyNameAttr.textContent == null)) {
-                verifyAttributeValue(it)
-            }
+            if (address != null) CommonDataTypeVerifier.verifyStringValues(address)
+            if (dNSName != null) CommonDataTypeVerifier.verifyStringValues(dNSName)
         }
     }
 
-    private fun verifyAttributeValue(it: Node) {
-        it.children("AttributeValue").forEach {
-            val nilAttribute = it.attributes.getNamedItemNS(TestCommon.XSI, "nil")?.textContent
-            if (StringUtils.isNotBlank(it.textContent) ||
-                    (nilAttribute != "true" && nilAttribute != "1"))
```

Comment:
I was able to verify that this is being covered by the schema validation, but I thought for sure it wouldn't because that technically means it is verifying the value of the `xsi:nil` attribute.